xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java

```diff
@@ -192,33 +192,19 @@ private void handleNewConfig(XdsConfig xdsConfig) {
     String newBalancerName = xdsConfig.newBalancerName;
     LbConfig childPolicy = xdsConfig.childPolicy;
     ManagedChannel lbChannel;
-    if (xdsLbState == null || !newBalancerName.equals(xdsLbState.balancerName)) {
-      if (xdsLbState != null) {
-        if (lrsWorking) {
-          lrsClient.stopLoadReporting();
-          lrsWorking = false;
-        }
-        ManagedChannel oldChannel = xdsLbState.shutdownAndReleaseChannel("Client shutdown");
-        oldChannel.shutdown();
-      }
-      try {
-        lbChannel = helper.createResolvingOobChannel(newBalancerName);
-      } catch (UnsupportedOperationException uoe) {
-        // Temporary solution until createResolvingOobChannel is implemented
-        // FIXME (https://github.com/grpc/grpc-java/issues/5495)
-        Logger logger = Logger.getLogger(XdsLoadBalancer.class.getName());
-        if (logger.isLoggable(FINEST)) {
-          logger.log(
-              FINEST,
-              "createResolvingOobChannel() not supported by the helper: " + helper,
-              uoe);
-          logger.log(
-              FINEST,
-              "creating oob channel for target {0} using default ManagedChannelBuilder",
-              newBalancerName);
-        }
-        lbChannel = ManagedChannelBuilder.forTarget(newBalancerName).build();
+    if (xdsLbState == null) {
+      lbChannel = initLbChannel(newBalancerName);
+      lrsClient =
```

Comment:
This can also be in `initLbChannel` to reduce duplicate code.

Reply:
Hmm. They really should be two pieces of logic. Having a method that creates the channel and `lrsClient` seems a little bit weird unless naming it `initLbChannelAndLrsClient()`.

Reply:
`init` might mean `create` and do something else.

Reply:
Another option is to move this line to near bottom, which also reduce duplicate.

Reply:
I would say, this ~1 line duplication is acceptable, rather than make the helper method do two separate things.